underfs/cos/src/main/java/alluxio/underfs/cos/COSInputStream.java

```diff
@@ -62,22 +68,42 @@
    * @param key the key of the file
    * @param client the client for COS
    * @param position the position to begin reading from
+   * @param retryPolicy the retry policy to use
    * @param multiRangeChunkSize the chunk size to use on this stream
+   *
    */
   COSInputStream(String bucketName, String key, COSClient client, long position,
-      long multiRangeChunkSize) throws IOException {
+      RetryPolicy retryPolicy, long multiRangeChunkSize) throws IOException {
     super(multiRangeChunkSize);
     mBucketName = bucketName;
     mKey = key;
     mCosClient = client;
     mPos = position;
+    mRetryPolicy = retryPolicy;
     ObjectMetadata meta = mCosClient.getObjectMetadata(mBucketName, key);
     mContentLength = meta == null ? 0 : meta.getContentLength();
   }
 
   @Override
   protected InputStream createStream(long startPos, long endPos)
       throws IOException {
+    // TODO(lu) only retry when object does not exist because of eventual consistency
+    if (mRetryPolicy == null) {
```

Comment:
i don't like these null checks

Reply:
Removed the null checks and using `CountingRetry(1)` instead, thanks